### PR TITLE
Replace Travis with Github Actions and Nix ❄️

### LINKED ISSUE #22: Replace Travis CI with Github Actions (Stack)
Since [some](https://github.com/arrow-kt/arrow/pull/1694/files) projects of 47 are moving to Github Actions, I think it would be a great idea to do the same!

Besides, I think it'd a great chance to **create our own Github action** for [Stack](https://docs.haskellstack.org/en/stable/README/) (since Github's one uses Cabal 3.0) and the rest of the Haskell community would benefit from it! 🎉 (maybe @gimenete can provide some guidance on how to do this 😉)

### PR DESCRIPTION
**Pull request by kutyel**
Huge thanks to @domenkozar for the help! 😉 🙌 

Fixes #22

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+on: [push]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: cachix/install-nix-action@v6
+    - uses: cachix/cachix-action@v2
+      with:
+        name: kutyel
```
Comment (pepegar) [inline .github/workflows/haskell.yml]: Does this mean we're publishing to cachix under `kutyel`?

**Comment by serras** [comment]
It seems, looking at the logs, that Nix uses a different root directory when compiling:

```
2019-12-17T16:00:50.5995315Z src/Mu/Adapter/ProtoBuf/Example.hs:1:1: error:
2019-12-17T16:00:50.5999118Z     Exception when trying to run compile-time code:
2019-12-17T16:00:50.6001605Z       test/protobuf/example.proto: openFile: does not exist (No such file or directory)
2019-12-17T16:00:50.6003989Z     Code: protobuf
2019-12-17T16:00:50.6016555Z             "ExampleProtoBufSchema" "test/protobuf/example.proto"
2019-12-17T16:00:50.6016676Z   |
2019-12-17T16:00:50.6017105Z 1 | {-# language DataKinds       #-}
```

File: default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  haskellnix = import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz);
+  overlay = _: pkgs:
+    let
+      hnPkgs = pkgs.haskell-nix.stackProject {
+        src = ./.;
+        #inherit stackYaml;
+        modules = [];
+      };
+    in {
+      mu-grpc-server = hnPkgs.mu-grpc-server.components.all;
```
Comment (serras) [inline default.nix]: Why `mu-grpc-server` explicitly?

File: .github/workflows/haskell.yml (hunk 1 above)
Comment (kutyel) [reply .github/workflows/haskell.yml]: I'm afraid yes, should I create a 47deg cachix account @juanpedromoreno? 😉

**Comment by domenkozar** [comment]
The downside of haskell.nix is that you have to compile GHC and everything up the to libraries first. It's a one time build you have to pay in order to get the benefit of not updating Nix when project goes on.

There's a way to use haskell.nix cachix to speed that up, but you'd need to pin the correct nixpkgs in that case (which may be ok with you?).

File: default.nix (hunk 2 above)
Comment (kutyel) [reply default.nix]: I have the same doubt 😂 @domenkozar? 😉

**Comment by kutyel** [comment]
> It seems, looking at the logs, that Nix uses a different root directory when compiling:
> 
> ```
> 2019-12-17T16:00:50.5995315Z src/Mu/Adapter/ProtoBuf/Example.hs:1:1: error:
> 2019-12-17T16:00:50.5999118Z     Exception when trying to run compile-time code:
> 2019-12-17T16:00:50.6001605Z       test/protobuf/example.proto: openFile: does not exist (No such file or directory)
> 2019-12-17T16:00:50.6003989Z     Code: protobuf
> 2019-12-17T16:00:50.6016555Z             "ExampleProtoBufSchema" "test/protobuf/example.proto"
> 2019-12-17T16:00:50.6016676Z   |
> 2019-12-17T16:00:50.6017105Z 1 | {-# language DataKinds       #-}
> ```

I just added some changes that should fix this, thanks again @domenkozar 🎉 

For reference, apparently cabal does not like external files, so you need to add them to your `data-files `entry on your cabal, I followed this [post](http://neilmitchell.blogspot.com/2008/02/adding-data-files-using-cabal.html) by Neil Mitchell 👏

**Comment by kutyel** [comment]
@juanpedromoreno it seems like the 47deg.cachix.org account is not correctly configured:
```json
responseBody = "{\"error\":\"No public key set yet. Run 'cachix generate-keypair' first.\"}"})
```
Can you please take a look? 😉

**Comment by kutyel** [comment]
Ok, it is fixed now 🎉  It should work! 🤞

**Pull request merged**